src/data/dfinsupp/ne_locus.lean

```diff
@@ -0,0 +1,120 @@
+  λ h, h ▸ by simp only [ne_locus, ne.def, eq_self_iff_true, not_true, finset.filter_false]⟩
+
+@[simp] lemma nonempty_ne_locus_iff {f g : Π₀ a, N a} : (f.ne_locus g).nonempty ↔ f ≠ g :=
+finset.nonempty_iff_ne_empty.trans ne_locus_eq_empty.not
+
+lemma ne_locus_comm : f.ne_locus g = g.ne_locus f :=
+by simp_rw [ne_locus, finset.union_comm, ne_comm]
+
+@[simp]
+lemma ne_locus_zero_right : f.ne_locus 0 = f.support :=
+by { ext, rw [mem_ne_locus, mem_support_iff, coe_zero, pi.zero_apply] }
+
+@[simp]
+lemma ne_locus_zero_left : (0 : Π₀ a, N a).ne_locus f = f.support :=
+(ne_locus_comm _ _).trans (ne_locus_zero_right _)
+
+end N_has_zero
+
+section ne_locus_and_maps
+
+variables {M P : α → Type*} [Π a, has_zero (N a)] [Π a, has_zero (M a)] [Π a, has_zero (P a)]
+
+lemma subset_map_range_ne_locus [Π a, decidable_eq (N a)] [Π a, decidable_eq (M a)]
+  (f g : Π₀ a, N a) {F : Π a, N a → M a} (F0 : ∀ a, F a 0 = 0) :
+  (f.map_range F F0).ne_locus (g.map_range F F0) ⊆ f.ne_locus g :=
+λ a, by simpa only [mem_ne_locus, map_range_apply, not_imp_not] using congr_arg (F a)
+
+lemma zip_with_ne_locus_eq_left [Π a, decidable_eq (N a)] [Π a, decidable_eq (P a)]
+  {F : Π a, M a → N a → P a} (F0 : ∀ a, F a 0 0 = 0)
+  (f : Π₀ a, M a) (g₁ g₂ : Π₀ a, N a) (hF : ∀ a f, function.injective (λ g, F a f g)) :
+  (zip_with F F0 f g₁).ne_locus (zip_with F F0 f g₂) = g₁.ne_locus g₂ :=
+by { ext, simpa only [mem_ne_locus] using (hF a _).ne_iff }
+
+lemma zip_with_ne_locus_eq_right [Π a, decidable_eq (M a)] [Π a, decidable_eq (P a)]
+  {F : Π a, M a → N a → P a} (F0 : ∀ a, F a 0 0 = 0)
+  (f₁ f₂ : Π₀ a, M a) (g : Π₀ a, N a) (hF : ∀ a g, function.injective (λ f, F a f g)) :
+  (zip_with F F0 f₁ g).ne_locus (zip_with F F0 f₂ g) = f₁.ne_locus f₂ :=
+by { ext, simpa only [mem_ne_locus] using (hF a _).ne_iff }
+
+lemma map_range_ne_locus_eq [Π a, decidable_eq (N a)] [Π a, decidable_eq (M a)] (f g : Π₀ a, N a)
+  {F : Π a, N a → M a} (F0 : ∀ a, F a 0 = 0) (hF : ∀ a, function.injective (F a)) :
+  (f.map_range F F0).ne_locus (g.map_range F F0) = f.ne_locus g :=
+by { ext, simpa only [mem_ne_locus] using (hF a).ne_iff }
+
+end ne_locus_and_maps
+
+section cancel_and_group
+variables [Π a, decidable_eq (N a)]
+
+@[simp] lemma add_ne_locus_add_eq_left [Π a, add_left_cancel_monoid (N a)] (f g h : Π₀ a, N a) :
+  (f + g).ne_locus (f + h) = g.ne_locus h  :=
+zip_with_ne_locus_eq_left _ _ _ _ $ λ a, add_right_injective
+
+@[simp] lemma add_ne_locus_add_eq_right [Π a, add_right_cancel_monoid (N a)] (f g h : Π₀ a, N a) :
+  (f + h).ne_locus (g + h) = f.ne_locus g  :=
+zip_with_ne_locus_eq_right _ _ _ _ $ λ a, add_left_injective
+
+variables [Π a, add_group (N a)] (f g h : Π₀ a, N a)
+
+@[simp] lemma neg_ne_locus_neg : (- f).ne_locus (- g) = f.ne_locus g :=
+map_range_ne_locus_eq _ _ (λ a, neg_zero) (λ a, neg_injective)
+
+lemma ne_locus_neg : (- f).ne_locus g = f.ne_locus (- g) :=
+by rw [← neg_ne_locus_neg _ _, neg_neg]
```

Comment:
The `neg` looks wrongly placed, unless you want `a.ne_locus b` to be referred to as `ne_locus_a_b` (which I think is very reasonable given that this is not an infix operation and that it matches `dist`), in which case you should rename the other lemmas in the file.